### ISSUE TITLE
How to define character varying(60) not null field?

### ISSUE THREAD

**Issue by fundon** [balderdashy/sails-postgresql#152]
Below does not work.

```js
username: {
  type: 'string',
  size: '60',
  required: true
}
```

**Comment by devinivy** [comment]
What column definition are you receiving instead?  You actually may be able to use `notNull: true` in your attribute definition.

**Comment by fundon** [comment]
@devinivy I checked the table schema

```
# \d+ users
...
username  | text
...
```

**Comment by devinivy** [comment]
Ah, it appears there's no `varchar` column type for postgresql: http://www.postgresql.org/docs/9.3/static/datatype.html#DATATYPE-TABLE .  So `size` does not appear to be supported for postgresql.

`text` should be correct, it seems.  Has your question been answered?

**Comment by fundon** [comment]
Sorry, I meant `character varying(60)`.

**Comment by devinivy** [comment]
Ah!  Actually this, crazily enough, is not supported!  But if you're waterline migration strategy is set to `safe`, you can change the schema manually without worrying about waterline complaining.

I'd love to see a PR to allow for `character` fields.  It shouldn't be too complicated!  See [here](https://github.com/balderdashy/sails-postgresql/blob/893e1c1349aab48138ad079ffa726bb11d109666/lib/utils.js#L61) and [here](https://github.com/balderdashy/sails-postgresql/blob/893e1c1349aab48138ad079ffa726bb11d109666/lib/utils.js#L225-L229) in the codebase if you're curious to implement this.

Want to change this issue to feature request?  If so, just re-title the issue, make a note in your first comment, and I'll add the appropriate label.  If you don't want to make this a feature request, I think we ought to close the issue out, unless you have any pending questions.

**Comment by fundon** [comment]
Okay, I'd love to submit a PR, :)

**Comment by devinivy** [comment]
Great!  Please do.  If you can make the change backward-compatible, it will likely make its way into the adapter.

**Comment by sailsbot** [comment]
Thanks for posting, @fundon.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
